### PR TITLE
derp: add a verifyClients check to the consistency check

### PR DESCRIPTION
**Pull request by raggi**
Only implemented for the local tailscaled variant for now.

Updates tailscale/corp#20844

### CODE REVIEW

File: derp/derp_server.go [hunk 1]
```diff
@@ -1955,6 +1955,22 @@ func (s *Server) ConsistencyCheck() error {
 			s.curClients.Value(),
 			len(s.clients)))
 	}
+	if s.verifyClientsLocalTailscaled {
```
Comment (maisem) [inline derp/derp_server.go:1958]: can you write a brief comment here to help readers understand what this is doing?

File: derp/derp_server.go [hunk 2]
```diff
@@ -1955,6 +1955,22 @@ func (s *Server) ConsistencyCheck() error {
...
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
```
Comment (maisem) [inline derp/derp_server.go:1959]: is second sufficient for the two calls? could `Status` be soo large that it fails?

File: derp/derp_server.go [hunk 3]
```diff
@@ -1955,6 +1955,22 @@ func (s *Server) ConsistencyCheck() error {
...
+		defer cancel()
+		status, err := localClient.Status(ctx)
```
Comment (maisem) [inline derp/derp_server.go:1961]: maybe StatusWIthoutPeers?

File: derp/derp_server.go [hunk 4]
```diff
@@ -1955,6 +1955,22 @@ func (s *Server) ConsistencyCheck() error {
...
+		if err != nil {
+			errs = append(errs, fmt.Sprintf("localClient.Status: %v", err))
+		}
+		info := &clientInfo{
+			IsProber: true,
+		}
+		clientIP := netip.IPv6Loopback()
+		if err := s.verifyClient(ctx, status.Self.PublicKey, info, clientIP); err != nil {
```
Comment (andrew-d) [inline derp/derp_server.go:1973]: Probably want to put this in an `else if` or you'll get a nil pointer exception if `Status` fails

**Review by maisem — APPROVED** [review]

File: derp/derp_server.go [hunk 5]
```diff
@@ -1955,6 +1955,22 @@ func (s *Server) ConsistencyCheck() error {
...
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+		defer cancel()
+		status, err := localClient.Status(ctx)
+		if err != nil {
+			errs = append(errs, fmt.Sprintf("localClient.Status: %v", err))
+		}
+		info := &clientInfo{
+			IsProber: true,
+		}
+		clientIP := netip.IPv6Loopback()
```
Comment (andrew-d) [inline derp/derp_server.go:1972]: Is there a particular reason that we use the IPv6 loopback addr here?

**Review by andrew-d — APPROVED** [review]

File: derp/derp_server.go (hunk 2 above)
Comment (raggi) [reply derp/derp_server.go:1963]: on a local socket? perhaps, seems unlikely though?

Comment (maisem) [reply derp/derp_server.go:1963]: I'm sure with large number of peers we would trigger this in just the JSON encoding/decoding

File: derp/derp_server.go (hunk 5 above)
Comment (maisem) [reply derp/derp_server.go:1972]: It's not even used on the other side so could just be netip.Addr{} I think

Comment (raggi) [reply derp/derp_server.go]: It was the first easy to reach for symbol, and happens to match what you'd generally get for local requests, and I didn't have any reason to plumb the web request data through.

File: derp/derp_server.go (hunk 4 above)
Comment (raggi) [reply derp/derp_server.go]: Moved out to a method

File: derp/derp_server.go (hunk 2 above)
Comment (raggi) [reply derp/derp_server.go]: Upped to 5s.

File: derp/derp_server.go (hunk 3 above)
Comment (raggi) [reply derp/derp_server.go]: done

**Review by andrew-d — APPROVED** [review]

File: derp/derp_server.go (hunk 5 above)
Comment (raggi) [reply derp/derp_server.go]: > It's not even used on the other side so could just be netip.Addr{} I think

True, I didn't want to leave a potential panic in a future change so I initialized both of the parameters.

**Pull request merged**